py/ai/h2o/sparkling/ml/features/H2OTargetEncoder.py

```diff
@@ -0,0 +1,108 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark.ml.util import JavaMLWritable
+from pyspark.ml.wrapper import JavaEstimator
+from pyspark.sql import SparkSession
+from pyspark import keyword_only
+from py_sparkling.ml.util import getValidatedEnumValue
+
+from pysparkling.context import H2OContext
+from pysparkling.spark_specifics import get_input_kwargs
+from ai.h2o.sparkling.ml.params import H2OTargetEncoderParams
+from py_sparkling.ml.util import set_double_values, JavaH2OMLReadable
+from ai.h2o.sparkling.ml.models import H2OTargetEncoderModel
+
+from h2o.utils.typechecks import assert_is_type, Enum
+
+
+class H2OTargetEncoder(H2OTargetEncoderParams, JavaEstimator, JavaH2OMLReadable, JavaMLWritable):
+
+    @keyword_only
+    def __init__(self, foldCol=None, labelCol="label", inputCols=[], holdoutStrategy = "None",
+                 blendedAvgEnabled=False, blendedAvgInflectionPoint=10.0, blendedAvgSmoothing=20.0, noise=0.01, noiseSeed=-1):
+        super(H2OTargetEncoder, self).__init__()
+        self._hc = H2OContext.getOrCreate(SparkSession.builder.getOrCreate(), verbose=False)
+        self._java_obj = self._new_java_obj("ai.h2o.sparkling.ml.features.H2OTargetEncoder", self.uid)
+
+        self._setDefault(foldCol=None, labelCol="label", inputCols=[], holdoutStrategy="None",
+                         blendedAvgEnabled=False, blendedAvgInflectionPoint=10.0, blendedAvgSmoothing=20.0, noise=0.01, noiseSeed=-1)
+        kwargs = get_input_kwargs(self)
+        self.setParams(**kwargs)
+
+
+    @keyword_only
+    def setParams(self, foldCol=None, labelCol="label", inputCols=[], holdoutStrategy = "None",
+                  blendedAvgEnabled=False, blendedAvgInflectionPoint=10.0, blendedAvgSmoothing=20.0, noise=0.01, noiseSeed=-1):
+        kwargs = get_input_kwargs(self)
+
+        if "holdoutStrategy" in kwargs:
+            kwargs["holdoutStrategy"] = getValidatedEnumValue(self.__getHoldoutStrategyEnumName(), kwargs["holdoutStrategy"])
```

Comment:
Just created helper method for this in master.

Just need to do `validatedEnumValue(self._H2OTargetEncoderParams__getHoldoutStrategyEnumName(), kwargs, "holdoutStrategy"`

Reply:
For reference: https://stackoverflow.com/questions/70528/why-are-pythons-private-methods-not-actually-private

Calling python private method from "outside" -> that is why the `_H2OTargetEncoderParams__` prefix

Reply:
Should I move the `__getHoldoutStrategyEnumName` method if to `H2OTargetEncoderParams` if setters are only in `TargetEncoder`?

Reply:
I probably don't understand :(.

Reply:
Setters are defined in `TargetEncoders` whereas other algos have them in `...Params`. I made update, please check whether it's ok?

Reply:
I see this is fine, thanks! :) Just question -> why are setters and getters separated? No need to change, maybe I'm just missing some benefit of that approach

Reply:
`H2OTargetEncoder`, `H2OTargetEncoderModel` and `H2OTargetEncoderMOJOModel` inherit from `H2OTargetEncoderParams`. Since parameters on models are read-only, setters are defined on `H2OTargetEncoder` and not on `H2OTargetEncoderParams`. If I'm not mistaken, Spark estimators use a similar design.

Reply:
great, thanks!